src/scripts/main.js

```diff
@@ -1,3 +1,280 @@
+    const select = document.createElement('select');
+
+    selectLabel.textContent = selectField.label;
+    select.name = selectField.name;
+    select.setAttribute('data-qa', selectField.data);
+    // select.required = true;
+
+    selectField.options.forEach((optionData) => {
+      const option = document.createElement('option');
+
+      option.value = optionData.value;
+      option.textContent = optionData.text;
+      select.appendChild(option);
+    });
+
+    selectLabel.appendChild(select);
+    form.appendChild(selectLabel);
+  }
+});
+
+const submitButton = document.createElement('button');
+
+submitButton.type = 'submit';
+submitButton.textContent = 'Save to table';
+form.appendChild(submitButton);
+
+document.body.appendChild(form);
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const data = {};
+
+  formData.forEach((value, key) => {
+    data[key] = value;
+  });
+
+  if (validation(data)) {
+    addNewRow(data);
+    pushNotification('Success', 'Row added successfully!', 'success');
+  }
+});
+
+function addNewRow(data) {
+  const tableBody = document.querySelector('tbody');
+  const newRow = document.createElement('tr');
+
+  for (const key in data) {
+    if (data.hasOwnProperty(key)) {
+      const newCell = document.createElement('td');
+
+      newCell.textContent = capitalizeWords(data[key]);
+
+      if (key === 'position') {
+        newCell.textContent = data[key];
+      }
+
+      if (key === 'salary') {
+        const salary = parseFloat(data[key]);
+
+        newCell.textContent = `$${salary.toLocaleString('en-US')}`;
+      }
+
+      newRow.appendChild(newCell);
+    }
+  }
+
+  tableBody.appendChild(newRow);
+}
+
+function capitalizeWords(text) {
+  return text
+    .toString()
+    .split(/[\s-]+/)
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+    .join(' ');
+}
+
+function validation(info) {
+  if (!info.name || !info.position || !info.office || !info.age || !info.salary) {
+    pushNotification('Missing Fields', 'All fields are required', 'error');
+
+    return false;
+  }
+
+  if (!info.name || info.name.trim().length < 4) {
+    pushNotification(
+      'Invalid Name',
+      'Name must have at least 4 characters.',
+      'error',
+    );
+
+    return false;
+  }
+
+  if (!info.position || info.position.trim() === '') {
+    pushNotification('Invalid Position', 'Position is required.', 'error');
+
+    return false;
+  }
+
+  if (!info.age || isNaN(info.age) || info.age < 18 || info.age > 90) {
+    pushNotification(
+      'Invalid Age',
+      'You must be older than 17 and younger than 91.',
+      'error',
+    );
+
+    return false;
+  }
+
+  if (!info.salary || isNaN(info.salary) || info.salary <= 0) {
+    pushNotification(
+      'Invalid Salary',
+      'Salary must be a positive number.',
+      'error',
+    );
+
+    return false;
+  }
```

Comment:
The salary validation logic requires a positive number. Ensure this requirement is communicated to users, possibly through placeholder text or form instructions.